send/netbox

```diff
@@ -0,0 +1,250 @@
+		or die("Can't open \$users_file_name\": $!\n");
+	local $/;
+	<$json_fh>
+};
+
+# Fetch Groups data from perun
+my $json_groups = do {
+	open(my $json_fh, "<:encoding(UTF-8)", $groups_file_name)
+		or die("Can't open \$groups_file_name\": $!\n");
+	local $/;
+	<$json_fh>
+};
+
+# Transform data from perun into hash
+my $perun_users = JSON->new->decode($json_users);
+my $perun_groups = JSON->new->decode($json_groups);
+
+# Fetch data from netbox
+my $netbox_users = call_server($USERS_API, $TYPE_GET, {})->{$RESULTS};
+my $netbox_groups = call_server($GROUPS_API, $TYPE_GET, {})->{$RESULTS};
+
+# Create hash of users for easier manipulation
+my $netbox_users_hash = {};
+for my $netbox_user (@{$netbox_users}) {
+	$netbox_users_hash->{$netbox_user->{"username"}} = $netbox_user;
+}
+
+# Create hash of groups for easier manipulation
+my $netbox_groups_hash = {};
+for my $netbox_group (@{$netbox_groups}) {
+	$netbox_groups_hash->{$netbox_group->{"name"}} = $netbox_group->{"id"};
+}
+
+# Create groups in netbox and associate all netbox groups ids with perun group names
+foreach (keys %$perun_groups) {
+	if (exists $netbox_groups_hash->{$_}) {
+		$perun_groups->{$_} = $netbox_groups_hash->{$_};
+	} else {
+		my $created_group = call_server($GROUPS_API, $TYPE_POST, {"name" => $_});
+		$perun_groups->{$created_group->{"name"}} = $created_group->{"id"};
+	}
+}
+
+# Create users in netbox
+foreach (keys %$perun_users) {
+	unless (exists $netbox_users_hash->{$_}) {
+		my $user = $perun_users->{$_};
+		my @alphanumeric = ('a'..'z', 'A'..'Z', 0..9);
+		my $randpassword = join '', map $alphanumeric[rand @alphanumeric], 0..10;
+		$user->{"password"} = $randpassword;
+		my @groups_ids = ();
+		for my $name (@{$user->{"groups"}}) {
+			my $group_id = $perun_groups->{$name};
+			push @groups_ids, int($group_id);
+		}
+		$user->{"groups"} = \@groups_ids;
+		call_server($USERS_API, $TYPE_POST, $user);
+	}
+}
+
+# Update existing netbox users
+foreach (keys %$netbox_users_hash) {
+	my @users_to_update = ();
+	if (exists $perun_users->{$_}) {
+		my $user = $perun_users->{$_};
+		$user->{"id"} = $netbox_users_hash->{$_}->{"id"};
+		my @groups_ids = ();
+		for my $name (@{$user->{"groups"}}) {
+			my $group_id = $perun_groups->{$name};
+			push @groups_ids, int($group_id);
+		}
+		$user->{"groups"} = \@groups_ids;
+		push @users_to_update, $user;
+	} else {
+		my $user = $netbox_users_hash->{$_};
+		$user->{"is_active"} = \0;
+		$user->{"groups"} = [];
+		push @users_to_update, $user;
+	}
+	call_server($USERS_API, $TYPE_PATCH, \@users_to_update);
+}
+
+###################### SUBROUTINES ######################
+
+#Name:
+# callServer
+#-----------------------
+#Parameters:
+# url     - url of server and method address to call
+# type    - GET, POST or PATCH
+# content - hash content of request, will be encoded to json output
+#-----------------------
+#Returns:
+# resolved status of call
+# 0 means everything is ok and done
```

Comment:
I would change this to something like: Returns hash of response content

Reply:
Fixed